### ISSUE TITLE
AWS VPC subnets are not tagged when using Kubernetes version 1.19

### ISSUE THREAD

**Issue by dbluxo** [weaveworks/eksctl#3341]
**What happened?**

When we create a cluster with version 1.18 using eksctl, in AWS the corresponding subnets are tagged with `kubernetes.io/cluster/<name>`.

If we change the version to 1.19 and create a cluster again (with same eksctl version), **the subnets are not tagged anymore.** This will cause the AWS  load balancer creation with nginx-ingress to no longer work, since it relies on the existence of the tags:

`Error syncing load balancer: failed to ensure load balancer: could not find any suitable subnets for creating the ELB`

**How to reproduce it?**

Our eksctl config file:

```
apiVersion: eksctl.io/v1alpha5
kind: ClusterConfig
metadata:
  name: eks-test
  version: "1.19"
  region: eu-central-1
vpc:
  nat:
    gateway: Disable
  subnets:
    private:
      eu-central-1a:
        id: subnet-11111111 # (anonymized)
      eu-central-1b:
        id: subnet-22222222 # (anonymized)
      eu-central-1c:
        id: subnet-33333333 # (anonymized)
  securityGroup: sg-4444444444444 # (anonymized)
  clusterEndpoints:
    publicAccess: false
    privateAccess: true

managedNodeGroups:
  - name: managed-core-eu-central-1a-blue
    ami: ami-123456789 # (anonymized)
    overrideBootstrapCommand: |
      /etc/eks/bootstrap.sh eks-test
    instanceType: t3.medium
    minSize: 1
    desiredCapacity: 1
    maxSize: 3
    availabilityZones:
      - eu-central-1a
    privateNetworking: true
    labels:
      type: core
    tags:
      Environment: Test
    ssh:
      allow: true
      publicKeyName: foo # (anonymized)
    iam:
      withAddonPolicies:
        externalDNS: true
        ebs: true
        autoScaler: true
        cloudWatch: false
      attachPolicyARNs:
        - arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly
        - arn:aws:iam::aws:policy/AmazonEKSWorkerNodePolicy
        - arn:aws:iam::aws:policy/AmazonEKS_CNI_Policy

  - name: managed-core-eu-central-1b-blue
    ami: ami-123456789 # (anonymized)
    overrideBootstrapCommand: |
      /etc/eks/bootstrap.sh eks-test
    instanceType: t3.medium
    minSize: 1
    desiredCapacity: 1
    maxSize: 3
    availabilityZones:
      - eu-central-1b
    privateNetworking: true
    labels:
      type: core
    tags:
      Environment: Test
    ssh:
      allow: true
      publicKeyName: foo # (anonymized)
    iam:
      withAddonPolicies:
        externalDNS: true
        ebs: true
        autoScaler: true
        cloudWatch: false
      attachPolicyARNs:
        - arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly
        - arn:aws:iam::aws:policy/AmazonEKSWorkerNodePolicy
        - arn:aws:iam::aws:policy/AmazonEKS_CNI_Policy

  - name: managed-core-eu-central-1c-blue
    ami: ami-123456789 # (anonymized)
    overrideBootstrapCommand: |
      /etc/eks/bootstrap.sh eks-test
    instanceType: t3.medium
    minSize: 1
    desiredCapacity: 1
    maxSize: 3
    availabilityZones:
      - eu-central-1c
    privateNetworking: true
    labels:
      type: core
    tags:
      Environment: Test
    ssh:
      allow: true
      publicKeyName: foo # (anonymized)
    iam:
      withAddonPolicies:
        externalDNS: true
        ebs: true
        autoScaler: true
        cloudWatch: false
      attachPolicyARNs:
        - arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly
        - arn:aws:iam::aws:policy/AmazonEKSWorkerNodePolicy
        - arn:aws:iam::aws:policy/AmazonEKS_CNI_Policy
```

**Versions**

```
$ eksctl version
0.38.0
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.3", GitCommit:"01849e73f3c86211f05533c2e807736e776fcf29", GitTreeState:"clean", BuildDate:"2021-02-18T12:10:55Z", GoVersion:"go1.15.8", Compiler:"gc", Platform:"darwin/amd64"}
```

**Comment by michaelbeaumont** [comment]
@dbluxo To be clear, you tested both `1.18` and `1.19` _with the same `eksctl` version_?

**Comment by dbluxo** [comment]
@michaelbeaumont

> To be clear, you tested both 1.18 and 1.19 with the same eksctl version?

Yes!

**Comment by bt909** [comment]
Interesting effect ist: cluster delployed with k8s 1.18, controlplane updated to 1.19 and then cluster removed (eksctl delete cluster...). The tags now are on the subnet and are not removed. I thought they should be removed.

**Comment by zezaeoh** [comment]
i found same issue in `ap-northeast-2`

**Comment by aclevername** [comment]
This is odd, I don't see how the introduction of 1.19 could of changed this behaviour, the PR to introduce it had a relatively small scope https://github.com/weaveworks/eksctl/pull/3285/files

**Comment by aclevername** [comment]
@zezaeoh @bt909 @dbluxo this does not appear to be due to any change in eksctl, but rather a change in the EKS api. eksctl never added this tag, it was previously getting added after the cluster was created by the EKS api: https://docs.aws.amazon.com/eks/latest/userguide/network_reqs.html#vpc-subnet-tagging
```
For 1.18 and earlier clusters, Amazon EKS adds the following tag to all subnets passed in during cluster creation. Amazon EKS does not add the tag to subnets passed in when creating 1.19 clusters
```

As of the latest AWS load balancer controller this tag is no longer needed (see 4th bullet point in [Prerequisites](https://docs.aws.amazon.com/eks/latest/userguide/alb-ingress.html)):
```
If you're using the AWS Load Balancer controller version v2.1.1 or earlier, subnets must be tagged as follows. If using version 2.1.2 or later, this tag is optional.
```

I don't know if the tag has any value outside of the AWS load balancer controller, if it does please share and we can introduce it into eksctl, other I think there is no change needed on the eksctl side and users should just update the AWS load balancer controller version

**Comment by dbluxo** [comment]
@aclevername :bulb: Thanks for the explanation! 

We use the [NGINX Ingress Controller](https://github.com/kubernetes/ingress-nginx), it needs these tags, otherwise we get the following errors:

```
Error syncing load balancer: failed to ensure load balancer: could not find any suitable subnets for creating the ELB
```

**Comment by aclevername** [comment]
> @aclevername  Thanks for the explanation!
> 
> We use the [NGINX Ingress Controller](https://github.com/kubernetes/ingress-nginx), it needs these tags, otherwise we get the following errors:
> 
> ```
> Error syncing load balancer: failed to ensure load balancer: could not find any suitable subnets for creating the ELB
> ```

Thanks @dbluxo thats good to know, in this case I think we have a few options from eksctl point of view, but the simplest is probably to have two actions:
1. Add this tag as a default for all subnets in new clusters, regardless of versions as its better that we manage this tag in cloudformation.
2. For existing clusters, we can add this tag as part of `eksctl upgrade cluster`  to provide a migration path to stop people from hitting the error you ran into.

**Comment by dbluxo** [comment]
Sorry I have to correct myself, the message does not come as a log from the NGINX Ingress Controller, but as a Kubernetes event (kubectl get events):

```
27s         Normal    EnsuringLoadBalancer     service/ingress-nginx-controller                Ensuring load balancer
63s         Normal    ScalingReplicaSet        deployment/ingress-nginx-controller             Scaled up replica set ingress-nginx-controller-bd6999c6b to 1
27s         Warning   SyncLoadBalancerFailed   service/ingress-nginx-controller                Error syncing load balancer: failed to ensure load balancer: could not find any suitable subnets for creating the ELB
```

We have configured the following in the nginx-ingress Kubernetes service:

```
apiVersion: v1
kind: Service
metadata:
  annotations:
    service.beta.kubernetes.io/aws-load-balancer-backend-protocol: http
    service.beta.kubernetes.io/aws-load-balancer-internal: 0.0.0.0/0
    service.beta.kubernetes.io/aws-load-balancer-ssl-cert: arn:aws:acm:eu-central-1:000000000000000 # (anonymized)
    service.beta.kubernetes.io/aws-load-balancer-ssl-ports: "443"
...
```

And apparently Kubernetes itself then goes and provides the load balancer in AWS.

The event seems to have originated here: https://github.com/kubernetes/kubernetes/blob/267e47f548ae2adedad2a78032f0982f4c7031e6/staging/src/k8s.io/legacy-cloud-providers/aws/aws.go#L3984

**Comment by keyboardfann** [comment]
From the EKS document, Amazon EKS seems don't adds the tags.

https://docs.amazonaws.cn/en_us/eks/latest/userguide/kubernetes-versions.html

Important

Starting with 1.19, Amazon EKS no longer adds the kubernetes.io/cluster/<cluster-name> tag to subnets passed in during cluster creation. This subnet tag is only required if you want to influence where the Kubernetes service controller or AWS Load Balancer Controller places Elastic Load Balancers. For more information about the requirements of subnets passed to Amazon EKS during cluster creation, see updates to Cluster VPC considerations.

Subnet tags are not modified on existing clusters updated to 1.19.

The AWS Load Balancer Controller version v2.1.1 and earlier required the <cluster-name> subnet tag. In version v2.1.2 and later, you can specify the tag to refine subnet discovery, but it's not required. For more information about the AWS Load Balancer Controller, see AWS Load Balancer Controller. For more information about subnet tagging when using a load balancer, see Application load balancing on Amazon EKS and Network load balancing on Amazon EKS.

**Comment by aclevername** [comment]
> Sorry I have to correct myself, the message does not come as a log from the NGINX Ingress Controller, but as a Kubernetes event (kubectl get events):
> 
> ```
> 27s         Normal    EnsuringLoadBalancer     service/ingress-nginx-controller                Ensuring load balancer
> 63s         Normal    ScalingReplicaSet        deployment/ingress-nginx-controller             Scaled up replica set ingress-nginx-controller-bd6999c6b to 1
> 27s         Warning   SyncLoadBalancerFailed   service/ingress-nginx-controller                Error syncing load balancer: failed to ensure load balancer: could not find any suitable subnets for creating the ELB
> ```
> 
> We have configured the following in the nginx-ingress Kubernetes service:
> 
> ```
> apiVersion: v1
> kind: Service
> metadata:
>   annotations:
>     service.beta.kubernetes.io/aws-load-balancer-backend-protocol: http
>     service.beta.kubernetes.io/aws-load-balancer-internal: 0.0.0.0/0
>     service.beta.kubernetes.io/aws-load-balancer-ssl-cert: arn:aws:acm:eu-central-1:000000000000000 # (anonymized)
>     service.beta.kubernetes.io/aws-load-balancer-ssl-ports: "443"
> ...
> ```
> 
> And apparently Kubernetes itself then goes and provides the load balancer in AWS.
> 
> The event seems to have originated here: https://github.com/kubernetes/kubernetes/blob/267e47f548ae2adedad2a78032f0982f4c7031e6/staging/src/k8s.io/legacy-cloud-providers/aws/aws.go#L3984
> 
> But it continues to be the case that the error occurs without the subnet tags and it works with subnet tags. Just wanted to clarify it.

Interesting, the code snippet you linked is different to what the latest `1.19.x` tag indicates is happening:

Error: https://github.com/kubernetes/kubernetes/blob/e19964183377d0ec2052d1f1fa930c4d7575bd50/staging/src/k8s.io/legacy-cloud-providers/aws/aws.go#L3934-L3936

Cause of error: https://github.com/kubernetes/kubernetes/blob/e19964183377d0ec2052d1f1fa930c4d7575bd50/staging/src/k8s.io/legacy-cloud-providers/aws/aws.go#L3392-L3477

It doesn't appear to be processing the `kubernetes.io/cluster/<name>` tag anymore. Are you sure your seeing this problem on a 1.19 cluster?

**Comment by dbluxo** [comment]
> It doesn't appear to be processing the kubernetes.io/cluster/<name> tag. Are you sure your seeing this problem on a 1.19 cluster?

Yes very sure, we encountered the problem only because we built a new cluster in version 1.19, before that we had no problems.

```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.3", GitCommit:"01849e73f3c86211f05533c2e807736e776fcf29", GitTreeState:"clean", BuildDate:"2021-02-18T12:10:55Z", GoVersion:"go1.15.8", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"19+", GitVersion:"v1.19.6-eks-49a6c0", GitCommit:"49a6c0bf091506e7bafcdb1b142351b69363355a", GitTreeState:"clean", BuildDate:"2020-12-23T22:10:21Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
```

Once I set the tag `kubernetes.io/cluster/<name>: shared` in the subnets manually, it works.

**Comment by aclevername** [comment]
> > It doesn't appear to be processing the kubernetes.io/cluster/ tag. Are you sure your seeing this problem on a 1.19 cluster?
> 
> Yes very sure, we encountered the problem only because we built a new cluster in version 1.19, before that we had no problems.
> 
> ```
> $ kubectl version
> Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.3", GitCommit:"01849e73f3c86211f05533c2e807736e776fcf29", GitTreeState:"clean", BuildDate:"2021-02-18T12:10:55Z", GoVersion:"go1.15.8", Compiler:"gc", Platform:"darwin/amd64"}
> Server Version: version.Info{Major:"1", Minor:"19+", GitVersion:"v1.19.6-eks-49a6c0", GitCommit:"49a6c0bf091506e7bafcdb1b142351b69363355a", GitTreeState:"clean", BuildDate:"2020-12-23T22:10:21Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
> ```
> 
> Once I set the tag `kubernetes.io/cluster/<name>: shared` in the subnets manually, it works.

Thanks for clarifying.

After digging some more I realise my initial statement was incorrect, the AWS cloud provider does still appear to be parsing this tag https://github.com/kubernetes/cloud-provider-aws/blob/0e8dc8ba69861e3fdfb0740b93e1068d08e1904f/pkg/providers/v2/tags.go#L93-L106

**Comment by aclevername** [comment]
> > > It doesn't appear to be processing the kubernetes.io/cluster/ tag. Are you sure your seeing this problem on a 1.19 cluster?
> > 
> > 
> > Yes very sure, we encountered the problem only because we built a new cluster in version 1.19, before that we had no problems.
> > ```
> > $ kubectl version
> > Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.3", GitCommit:"01849e73f3c86211f05533c2e807736e776fcf29", GitTreeState:"clean", BuildDate:"2021-02-18T12:10:55Z", GoVersion:"go1.15.8", Compiler:"gc", Platform:"darwin/amd64"}
> > Server Version: version.Info{Major:"1", Minor:"19+", GitVersion:"v1.19.6-eks-49a6c0", GitCommit:"49a6c0bf091506e7bafcdb1b142351b69363355a", GitTreeState:"clean", BuildDate:"2020-12-23T22:10:21Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
> > ```
> > 
> > 
> > Once I set the tag `kubernetes.io/cluster/<name>: shared` in the subnets manually, it works.
> 
> Thanks for clarifying.
> 
> After digging some more I realise my initial statement was incorrect, the AWS cloud provider does still appear to be parsing this tag https://github.com/kubernetes/kubernetes/blob/e19964183377d0ec2052d1f1fa930c4d7575bd50/staging/src/k8s.io/legacy-cloud-providers/aws/aws.go#L3367-L3372

@dbluxo I just created a 1.19 cluster, and provisioned the ingress controller following [the docs](https://kubernetes.github.io/ingress-nginx/deploy/#aws). I managed to successfully provision an ingress resource without the subnet tag. Is it possible you are running an older version of nginx ingress controller?

**Comment by aclevername** [comment]
@dbluxo Thanks.

Can you confirm what `kubernetes.io` tags you have on the subnets? I want to check if that `kubernetes.io/role/elb` / `kubernetes.io/role/internal-elb` tag exists https://aws.amazon.com/premiumsupport/knowledge-center/eks-vpc-subnet-discovery/

**Comment by dbluxo** [comment]
@aclevername We have:

- a lot of `kubernetes.io/cluster/<name>: shared` tags
- one `kubernetes.io/role/internal-elb: <subnet_name>` tag

**Comment by aclevername** [comment]
> @aclevername We have:
> 
> * a lot of `kubernetes.io/cluster/<name>: shared` tags
> * one `kubernetes.io/role/internal-elb: <subnet_name>` tag

thanks. Just to make sure I have the full picture as I'm quite puzzled by whats causing this:
- fresh 1.19 cluster
- nginx ingress controller installed
- `kubernetes.io/role/internal-elb` subnet tag exists
- creating an ingress resource fails as it cannot provision the load balancer, but it succeeds when you add the `kubernetes.io/cluster/<name>` tag

Are you running the AWS load balancer controller as well? Or anything else that would be response for provisoning load balancers? It should work when using the built in k8s cloud-provider for AWS

**Comment by dbluxo** [comment]
> thanks @dbluxo . Just to make sure I have the full picture as I'm quite puzzled by whats causing this:
> 
>     * fresh 1.19 cluster
> 
>     * nginx ingress controller installed
> 
>     * `kubernetes.io/role/internal-elb` subnet tag exists
> 
>     * creating an ingress resource fails as it cannot provision the load balancer, but it succeeds when you add the `kubernetes.io/cluster/<name>` tag

Yes, exactly.

> Are you running the AWS load balancer controller as well? 

No

> Or anything else that would be response for provisoning load balancers?

No

Does the tag with the key `kubernetes.io/role/internal-elb` have to have a specific value?

**Comment by aclevername** [comment]
> Does the tag with the key `kubernetes.io/role/internal-elb` have to have a specific value?

The docs aren't very clear, but it looks like they just use `1` as the value https://aws.amazon.com/premiumsupport/knowledge-center/eks-vpc-subnet-discovery/
```
For private subnets used by internal load balancers

To allow Kubernetes to use your private subnets for internal load balancers, tag all private subnets in your VPC with the following key-value pair:

Key: kubernetes.io/role/internal-elb
Value: 1
For public subnets used by external load balancers

To allow Kubernetes to use only tagged subnets for external load balancers, tag all public subnets in your VPC with the following key-value pair:

Key: kubernetes.io/role/elb
Value: 1
```

**Comment by dbluxo** [comment]
> The docs aren't very clear, but it looks like they just use `1` as the value https://aws.amazon.com/premiumsupport/knowledge-center/eks-vpc-subnet-discovery/
> 
> ```
> For private subnets used by internal load balancers
> 
> To allow Kubernetes to use your private subnets for internal load balancers, tag all private subnets in your VPC with the following key-value pair:
> 
> Key: kubernetes.io/role/internal-elb
> Value: 1
> For public subnets used by external load balancers
> 
> To allow Kubernetes to use only tagged subnets for external load balancers, tag all public subnets in your VPC with the following key-value pair:
> 
> Key: kubernetes.io/role/elb
> Value: 1
> ```

Thanks @aclevername, changed it to `1`, unfortunately does not work with it either.

**Comment by aclevername** [comment]
> > The docs aren't very clear, but it looks like they just use `1` as the value https://aws.amazon.com/premiumsupport/knowledge-center/eks-vpc-subnet-discovery/
> > ```
> > For private subnets used by internal load balancers
> > 
> > To allow Kubernetes to use your private subnets for internal load balancers, tag all private subnets in your VPC with the following key-value pair:
> > 
> > Key: kubernetes.io/role/internal-elb
> > Value: 1
> > For public subnets used by external load balancers
> > 
> > To allow Kubernetes to use only tagged subnets for external load balancers, tag all public subnets in your VPC with the following key-value pair:
> > 
> > Key: kubernetes.io/role/elb
> > Value: 1
> > ```
> 
> Thanks @aclevername, changed it to `1`, unfortunately does not work with it either.

Thanks @dbluxo . I'm sorry I haven't been able to reproduce this, I think its clear this issue isn't related to eksctl. I think the best action would be to open an issue on https://github.com/kubernetes/cloud-provider-aws with the details you've provided, as the error message you got earlier ` Error syncing load balancer: failed to ensure load balancer: could not find any suitable subnets for creating the ELB` is coming from the AWS cloud provider. Hopefully they will be able to help you figure out whats going on :smile:

**Comment by bt909** [comment]
> > > The docs aren't very clear, but it looks like they just use `1` as the value https://aws.amazon.com/premiumsupport/knowledge-center/eks-vpc-subnet-discovery/
> > > ```
> > > For private subnets used by internal load balancers
> > > 
> > > To allow Kubernetes to use your private subnets for internal load balancers, tag all private subnets in your VPC with the following key-value pair:
> > > 
> > > Key: kubernetes.io/role/internal-elb
> > > Value: 1
> > > For public subnets used by external load balancers
> > > 
> > > To allow Kubernetes to use only tagged subnets for external load balancers, tag all public subnets in your VPC with the following key-value pair:
> > > 
> > > Key: kubernetes.io/role/elb
> > > Value: 1
> > > ```
> > 
> > 
> > Thanks @aclevername, changed it to `1`, unfortunately does not work with it either.
> 
> Thanks @dbluxo . I'm sorry I haven't been able to reproduce this, I think its clear this issue isn't related to eksctl. I think the best action would be to open an issue on https://github.com/kubernetes/cloud-provider-aws with the details you've provided, as the error message you got earlier ` Error syncing load balancer: failed to ensure load balancer: could not find any suitable subnets for creating the ELB` is coming from the AWS cloud provider. Hopefully they will be able to help you figure out whats going on 😄

Hello @aclevername,

you can't reproduce this? You are using custom VPC with private subnets and your nginx ingress controller creates the load balancer on a private subnet?
Interesting, as it seems that this tag creation comes from AWS itself:
https://docs.aws.amazon.com/eks/latest/userguide/kubernetes-versions.html (Kubernetes 1.19).

And yes, you are right, it seems not a eksctl related issue (or should it be worth to add that feature (tagging of subnets))?.

**Comment by aclevername** [comment]
> Hello @aclevername,
> 
> you can't reproduce this? You are using custom VPC with private subnets and your nginx ingress controller creates the load balancer on a private subnet?

Hey @bt909 :wave:  I created a fresh 1.19 cluster using eksctl `eksctl create custer --name jk-19 --version 1.19 --managed`, observed no `kubernetes.io/cluster/<name>` tag on the subnet, but was able to provision load balancers without errors. Both using the NGINX ingress controller, and creating the type `LoadBalancer` service manually.

> Interesting, as it seems that this tag creation comes from AWS itself:
> https://docs.aws.amazon.com/eks/latest/userguide/kubernetes-versions.html (Kubernetes 1.19).

Yes this tag used to get added by eks in versions prior to 1.19, but it no longer gets added in 1.19. Existing clusters that get upgraded will retain the tag as its not removed. But this tag should *not be required* according to AWS.

> And yes, you are right, it seems not a eksctl related issue (or should it be worth to add that feature (tagging of subnets))?.

I think this depends on what the cause of the problem is. Supposedly this tag isn't required, so adding the functionality to eksctl as a workaround to a bug is less than ideal, I'd rather the bug was fixed on EKS side. In the meantime users can force this tag to exist on the subnet by adding it manually or using the [tags](https://eksctl.io/usage/schema/#metadata-tags) functionality to add this tag to all resources eksctl creates (not ideal, but at least a workaround for now). 

If this isn't a bug but a mistake in the EKS docs then yes I think we would make the changes in eksctl to enforce the adding of this tag, like we do currently with the `kubernetes.io/role/elb`.

**Comment by dbluxo** [comment]
> I think this depends on what the cause of the problem is. Supposedly this tag isn't required, so adding the functionality to eksctl as a workaround to a bug is less than ideal, I'd rather the bug was fixed on EKS side. In the meantime users can force this tag to exist on the subnet by adding it manually or using the tags functionality to add this tag to all resources eksctl creates (not ideal, but at least a workaround for now).

> If this isn't a bug but a mistake in the EKS docs then yes I think we would make the changes in eksctl to enforce the adding of this tag, like we do currently with the `kubernetes.io/role/elb`.

@aclevername  I'm not sure right now which way we could find out if it's a bug or not?

**Comment by dbluxo** [comment]
@aclervername I was not sure because you wrote that it could be fixed on EKS side, but the AWS cloud provider is not EKS but directly a part of Kubernetes?

Anyway, I have opened an [issue](https://github.com/kubernetes/cloud-provider-aws/issues/184), you can look over it and possibly give further input :blush:

**Comment by aclevername** [comment]
> @aclervername I was not sure because you wrote that it could be fixed on EKS side, but the AWS cloud provider is not EKS but directly a part of Kubernetes? 


My understanding is that EKS overrides the built-in aws cloud-provider with a patched version, so if there's a bug in it they could fix it, or bring back the automatic tagging :smile:. For example they patched the cloud provider in 1.19: https://github.com/aws/eks-distro/blob/main/projects/kubernetes/kubernetes/1-19/patches/0010-EKS-PATCH-additional-subnet-configuration-for-AWS-el.patch

> Anyway, I have opened an [issue](https://github.com/kubernetes/cloud-provider-aws/issues/184), you can look over it and possibly give further input

Will do!
